iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/vector.inl

```diff
@@ -249,26 +254,33 @@ inline bool vector<T, Capacity>::resize(const uint64_t count, const Targs&... ar
 template <typename T, uint64_t Capacity>
 inline T* vector<T, Capacity>::data() noexcept
 {
+    // AXIVION Next Construct AutosarC++19_03-A5.2.4 : Type-safety ensured by template parameter
+    // NOLINTNEXTLINE (cppcoreguidelines-pro-type-reinterpret-cast)
     return reinterpret_cast<T*>(m_data);
 }
 
 template <typename T, uint64_t Capacity>
 inline const T* vector<T, Capacity>::data() const noexcept
 {
+    // AXIVION Next Construct AutosarC++19_03-A5.2.4 : Type-safety ensured by template parameter
+    // NOLINTNEXTLINE (cppcoreguidelines-pro-type-reinterpret-cast)
```

Comment:
We should use here:
```cxx
return &at(0);
```
Then we do not need to add an axivion and clang-tidy exception.

Reply:
I used that in the const `data()` and called that one from here.